docs/intro/upgrading-to-ionic-6.md

```diff
@@ -0,0 +1,443 @@
+# Upgrading to Ionic 6
+
+Follow this guide to upgrade your Ionic 5 apps to Ionic 6.
+
+## Getting Started
+
+### Angular
+
+1. Ionic 6 supports Angular 12+. Update to the latest version of Angular by following the [Angular Update Guide](https://update.angular.io/).
+2. Update to the latest version of Ionic 6:
+
+```shell
+npm install @ionic/angular@6
+```
+
+If you are using Ionic Angular Server, be sure to update that as well:
+
+```shell
+npm install @ionic/angular@6 @ionic/angular-server@6
+```
+
+3. Remove any usage of `Config.set()`. Instead, set your config in `IonicModule.forRoot()`. See the [Angular Config Documentation](../angular/config) for more examples.
+4. Remove any usage of the `setupConfig` function previously exported from `@ionic/angular`. Set your config in `IonicModule.forRoot()` instead.
+
+### React
+
+1. Ionic 6 supports React 17+. Update to the latest version of React:
+
+```shell
+npm install react@latest react-dom@latest
+```
+
+2. Update to the latest version of Ionic 6:
+
+```shell
+npm install @ionic/react@6 @ionic/react-router@6
+```
+
+3. Update the `test` field in the `scripts` object of your `package.json` to include `transformIgnorePatterns`:
+
+```json
+"scripts": {
+  "test": "react-scripts test --transformIgnorePatterns 'node_modules/(?!(@ionic/react|@ionic/react-router|@ionic/core|@stencil/core|ionicons)/)'",
+  ...
+}
+```
+
+4. Import and call `setupIonicReact` in your `App` component file. If you are also using `setupConfig`, pass your config to `setupIonicReact` instead:
+
+**Before**
+```tsx
```

Comment:
Docusaurus supports file names when specifying the language:

```
```tsx title="App.tsx"
```

This just cleans up the copied code for the implementer. 

This would apply to all places we are doing:
```
// App.tsx
```

as well as for possibly the Jest config and package.json call outs.